conformance/lang/expressions/literals/int_literal.balt

```diff
@@ -0,0 +1,478 @@
+Test-Case: output
+Description: Test DecimalNumbers with int-type-descriptor.
+Labels: int
+
+public function main() {
+    int x = 101;
+    io:println(x); // @output 101
+
+    x = 0;
+    io:println(x); // @output 0
+
+    x = 3;
+    io:println(x); // @output 3
+
+    x = 9223372036854775807;
+    io:println(x); // @output 9223372036854775807
+}
+
+Test-Case: output
+Description: Test DecimalNumbers with function call.
+Labels: function-call-expr, int
+
+public function main() {
+    printInt(101); // @output 101
+
+    printInt(0); // @output 0
+
+    printInt(3); // @output 3
+
+    printInt(9223372036854775807); // @output 9223372036854775807
+}
+
+function printInt(int x) {
+    io:println(x);
+}
+
+Test-Case: error
+Description: Test DecimalNumbers with leading zeros.
+Labels: int
+
+public function main() {
+    int x = 00; // @error invalid DecimalNumber with leading '0'
+
+    x = 012345; // @error invalid DecimalNumber with leading '0'
+}
+
+Test-Case: output
+Description: Test HexIntLiterals with int-type-descriptor.
+Labels: function-call-expr, HexIntLiteral, int
+
+public function main() {
+    int x = 0x1;
+    printInt(x); // @output 1
+
+    x = 0xa;
+    printInt(x); // @output 10
+
+    x = 0xA;
+    printInt(x); // @output 10
+
+    x = 0x000000a;
+    printInt(x); // @output 10
+
+    x = 0x1fed12cba7;
+    printInt(x); // @output 137121418151
+
+    x = 0xABCDEF81;
+    printInt(x); // @output 2882400129
+
+    x = 0x192AAce;
+    printInt(x); // @output 26389198
+
+    x = 0X1;
+    printInt(x); // @output 1
+
+    x = 0Xa;
+    printInt(x); // @output 10
+
+    x = 0XA;
+    printInt(x); // @output 10
+
+    x = 0XABCDEF81;
+    printInt(x); // @output 2882400129
+
+    printInt(0X1fedcba7); // @output 535677863
+
+    printInt(0X192AAc5); // @output 26389189
+}
+
+function printInt(int x) {
+    io:println(x);
+}
+
+Test-Case: parser-error
+Description: Test invalid HexInLiteral.
+Fail-Issue: ballerina-lang#32559
+Labels: int
+
+public function main() {
+    int x = 0x; // @error invalid HexIntLiteral due to absence of HexDigits
+
+    x = 0X; // @error invalid HexIntLiteral due to absence of HexDigits
+
+    x = 0x123g; // @error invalid HexIntLiteral due to incorrect HexDigits
+
+    x = 0X123G; // @error invalid HexIntLiteral due to incorrect HexDigits
+
+    x = 123ab; // @error undefined symbol 'ab'
+}
+
+Test-Case: output
+Description: Test int-literals with toString().
+Fail-Issue: ballerina-lang#32769
+Labels: DecimalNumber, HexIntLiteral, value:toString
+
+public function main() {
+    io:println((101).toString()); // @output 101
+
+    io:println(0.toString()); // @output 0
+
+    io:println(9223372036854775807.toString()); // @output 9223372036854775807
+
+    io:println(0x12ab.toString()); // @output 4779
```

Comment:
Only this fails here right? Shall we move this out to a separate test?

Reply:
We can split these tests into two anyway. DecimalNumber vs HexIntLiteral.

Reply:
Will update.